applications/IgaApplication/custom_modelers/iga_modeler_sbm.cpp

```diff
@@ -0,0 +1,601 @@
+    }
+
+    std::string quadrature_method = rParameters.Has("quadrature_method")
+        ? rParameters["integration_rule"].GetString()
+        : "GAUSS";
+
+    KRATOS_INFO_IF("CreateQuadraturePointGeometries", mEchoLevel > 0)
+        << "Creating " << name << "s of type: " << type
+        << " for " << rGeometryList.size() << " geometries"
+        << " in " << rModelPart.Name() << "-SubModelPart." << std::endl;
+
+    // Check if the sbm projection operation is needed (there is no need for background domain and for body-fitted boundary conditions)
+    PointVector points;   
+    const std::string skin_model_part_name = mParameters.Has("skin_model_part_name")
+            ? mParameters["skin_model_part_name"].GetString()
+            : "skin_model_part";
+
+    ModelPart& skin_model_part = mpModel->HasModelPart(skin_model_part_name)
+            ? mpModel->GetModelPart(skin_model_part_name)
+            : KRATOS_ERROR << "::[CreateQuadraturePointGeometriesSbm]::: Sbm case -> skin_model_part has not been defined before. "
+                            << "Maybe you are not calling the nurbs_modeler_sbm" << std::endl;
+
+    // inner & outer are defaulf sub model part names
+    ModelPart& skin_sub_model_part_in = skin_model_part.GetSubModelPart("inner");
+    ModelPart& skin_sub_model_part_out = skin_model_part.GetSubModelPart("outer");
+
+    const bool is_inner = rParameters["sbm_parameters"]["is_inner"].GetBool();
+    
+    const std::string surrogate_sub_model_part_name = is_inner ? "surrogate_inner" : "surrogate_outer";
+    
+    ModelPart& surrogate_sub_model_part = rModelPart.GetParentModelPart().HasSubModelPart(surrogate_sub_model_part_name)
+            ? rModelPart.GetParentModelPart().GetSubModelPart(surrogate_sub_model_part_name)
+            : KRATOS_ERROR << "::[CreateQuadraturePointGeometriesSbm]::: Sbm case -> surrogate_sub_model_part has not been defined before."
+                            << "Maybe you are not calling the nurbs_modeler_sbm" << std::endl;
+
+    if (is_inner) { // INNER
+        for (auto &i_cond : skin_sub_model_part_in.Conditions()) {
+            points.push_back(PointTypePointer(new PointType(i_cond.Id(), i_cond.GetGeometry().Center().X(), i_cond.GetGeometry().Center().Y(), i_cond.GetGeometry().Center().Z())));
+        }
+    } 
+    else { // OUTER
+        for (auto &i_cond : skin_sub_model_part_out.Conditions()) {
+            points.push_back(PointTypePointer(new PointType(i_cond.Id(), i_cond.GetGeometry().Center().X(), i_cond.GetGeometry().Center().Y(), i_cond.GetGeometry().Center().Z())));
+        }
+    }
+    
+    // Get the mesh sizes from the surrogate model part
+    const Vector knot_span_sizes = surrogate_sub_model_part.GetParentModelPart().GetValue(KNOT_SPAN_SIZES);
+    // Get the parameter space corners from the surrogate model part
+    const std::vector<Vector> parameter_space_corners = surrogate_sub_model_part.GetParentModelPart().GetValue(PARAMETER_SPACE_CORNERS);
```

Comment:
```suggestion
    const Vector& knot_span_sizes = surrogate_sub_model_part.GetParentModelPart().GetValue(KNOT_SPAN_SIZES);
    // Get the parameter space corners from the surrogate model part
    const std::vector<Vector>& parameter_space_corners = surrogate_sub_model_part.GetParentModelPart().GetValue(PARAMETER_SPACE_CORNERS);
```
Can't these be references?